src/System.Windows.Forms/src/System/Windows/Forms/ListBox.cs

```diff
@@ -3890,6 +3923,730 @@ public class SelectedObjectCollection : IList {
                 }
             }
         }
+
+        internal override bool SupportsUiaProviders => true;
+
+        /// <devdoc>
+        /// </devdoc>
+        protected override AccessibleObject CreateAccessibilityInstance()
+        {
+            return new ListBoxAccessibleObject(this);
+        }
+
+        /// <summary>
+        /// ListBox control accessible object with UI Automation provider functionality.
+        /// This inherits from the base ListBoxExAccessibleObject and ListBoxAccessibleObject
+        /// to have all base functionality.
+        /// </summary>
+        [ComVisible(true)]
+        internal class ListBoxAccessibleObject : ControlAccessibleObject
+        {
+            private readonly ListBox _owningListBox;
+            private readonly ListBoxItemAccessibleObjectCollection _itemAccessibleObjects;
+            private readonly IAccessible _systemIAccessible;
+
+            /// <summary>
+            /// Initializes new instance of ListBoxAccessibleObject.
+            /// </summary>
+            /// <param name="owningListBox">The owning ListBox control.</param>
+            public ListBoxAccessibleObject(ListBox owningListBox) : base(owningListBox)
+            {
+                _owningListBox = owningListBox;
+                _itemAccessibleObjects = new ListBoxItemAccessibleObjectCollection(owningListBox, this);
+                _systemIAccessible = GetSystemIAccessibleInternal();
+            }
+
+            public override AccessibleObject GetChild(int index)
+            {
+                return GetChildFragment(index);
+            }
+
+            public override string DefaultAction => "";/*_systemIAccessible?.accDefaultAction[0];*/ //TODO need write action
+
+            public override AccessibleStates State
+            {
+                get
+                {
+                    AccessibleStates state = AccessibleStates.Focusable;
+
+                    //if (_owningListBox.Focused)
+                    //{
+                    //    state |= AccessibleStates.Focused;  //This does not work. Anyway ListBoxAccessibleObject may have Focused state
+                    //}
+                    return state;
+                }
+            }
+
+            internal override UnsafeNativeMethods.IRawElementProviderFragment GetFocus()
+            {
+                Debug.WriteLine("Get focus LB");
+                return GetChildFragment(_owningListBox.SelectedIndex);
+            }
+
+            public override AccessibleObject GetFocused()
+            {
+                return GetChildFragment(_owningListBox.SelectedIndex);
```

Comment:
GetFocused can call GetFocus or vice versa.
```suggestion
                return GetFocus();
```

Reply:
Thanks, at the moment GetFocus calls GetFocused:
https://github.com/dotnet/winforms/blob/0db05240631f73edd4f4da29308104623888ce85/src/System.Windows.Forms/src/System/Windows/Forms/ListBox.cs#L4133-L4136
and GetFocused has this implementation:
https://github.com/dotnet/winforms/blob/0db05240631f73edd4f4da29308104623888ce85/src/System.Windows.Forms/src/System/Windows/Forms/ListBox.cs#L4245-L4254